python/lsst/cp/pipe/defects.py

```diff
@@ -0,0 +1,645 @@
+           'FindDefectsTaskConfig', ]
+
+import numpy as np
+import matplotlib.pyplot as plt
+from matplotlib.backends.backend_pdf import PdfPages
+import os
+import warnings
+
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+import lsst.afw.image as afwImage
+import lsst.meas.algorithms as measAlg
+import lsst.afw.math as afwMath
+import lsst.afw.detection as afwDetection
+import lsst.log as lsstLog
+import lsst.afw.display as afwDisplay
+import lsst.afw.cameraGeom.utils as cgUtils
+
+from lsst.ip.isr import IsrTask
+from .utils import NonexistentDatasetTaskDataIdContainer, SingleVisitListTaskRunner, countMaskedPixels, \
+    validateIsrConfig
+
+
+class FindDefectsTaskConfig(pexConfig.Config):
+    """Config class for defect finding"""
+
+    isrForFlats = pexConfig.ConfigurableField(
+        target=IsrTask,
+        doc="""Task to perform instrumental signature removal""",
+    )
+    isrForDarks = pexConfig.ConfigurableField(
+        target=IsrTask,
+        doc="""Task to perform instrumental signature removal""",
+    )
+    isrMandatoryStepsFlats = pexConfig.ListField(
+        dtype=str,
+        doc="isr operations that must be performed for valid results when using flats. " +
+        "Raises if any of these are False",
+        default=['doAssembleCcd', 'doFringe']
+    )
+    isrMandatoryStepsDarks = pexConfig.ListField(
+        dtype=str,
+        doc="isr operations that must be performed for valid results when using darks. " +
+        "Raises if any of these are False",
+        default=['doAssembleCcd', 'doFringe']
+    )
+    isrForbiddenStepsFlats = pexConfig.ListField(
+        dtype=str,
+        doc="isr operations that must NOT be performed for valid results when using flats. " +
+        "Raises if any of these are True",
+        default=['doAddDistortionModel', 'doBrighterFatter', 'doUseOpticsTransmission',
+                 'doUseFilterTransmission', 'doUseSensorTransmission', 'doUseAtmosphereTransmission']
+    )
+    isrForbiddenStepsDarks = pexConfig.ListField(
+        dtype=str,
+        doc="isr operations that must NOT be performed for valid results when using darks. " +
+        "Raises if any of these are True",
+        default=['doAddDistortionModel', 'doBrighterFatter', 'doUseOpticsTransmission',
+                 'doUseFilterTransmission', 'doUseSensorTransmission', 'doUseAtmosphereTransmission']
+    )
+    isrDesirableSteps = pexConfig.ListField(
+        dtype=str,
+        doc="isr operations that it is advisable to perform, but are not mission-critical." +
+        " WARNs are logged for any of these found to be False.",
+        default=['doBias']
+    )
+    ccdKey = pexConfig.Field(
+        dtype=str,
+        doc="The key by which to pull a detector from a dataId, e.g. 'ccd' or 'detector'",
+        default='ccd',
+    )
+    imageTypeKey = pexConfig.Field(
+        dtype=str,
+        doc="The key by which to check whether images are darks or flats",
```

Comment:
It's the key by which to query the exposure via the butler, _i.e._ with `butler.queryMetadata('raw', self.config.imageTypeKey, dataId=dataId)`. It might be standard, but I believe this can be different in different obs_packages.